CI: Integration Tests - TestRecomputeIdentityPolicy

### Test Name

TestRecomputeIdentityPolicy

### Test Failure Output

```shell
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/policy/compute   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestRecomputeIdentityPolicy (0.01s)
--- FAIL: TestRecomputeIdentityPolicy/creates_entry_and_fires_waiting_watch (0.00s)

    logger.go:256: time=2026-07-10T14:36:16.234Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:403 msg="Starting hive"
    logger.go:256: time=2026-07-10T14:36:16.235Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:407 msg="Started hive" duration=109.722µs
    compute_test.go:38: 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/policy/compute/compute_test.go:38
        	Error:      	Should be false
        	Test:       	TestRecomputeIdentityPolicy/creates_entry_and_fires_waiting_watch
    logger.go:256: time=2026-07-10T14:36:16.235Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:421 msg="Stopping hive"
    logger.go:256: time=2026-07-10T14:36:16.235Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:425 msg="Stopped hive" duration=94.57µs
```

 ⛑️ The following owners are responsible for reliability of the failing test(s): 
Package                                       Owner
github.com/cilium/cilium/pkg/policy/compute   @cilium/sig-policy

--- a/pkg/policy/compute/compute_test.go
+++ b/pkg/policy/compute/compute_test.go
@@ -28,11 +28,10 @@ func TestRecomputeIdentityPolicy(t *testing.T) {
 	testutils.GoleakVerifyNone(t, testutils.GoleakIgnoreCurrent())
 
 	t.Run("creates entry and fires waiting watch", func(t *testing.T) {
-		_, _, computer, idmgr := fixture(t)
+		_, _, computer, _ := fixture(t)
 
 		targetID := identity.NumericIdentity(7)
 		id := identity.NewIdentity(targetID, labels.Labels{})
-		idmgr.Add(id)
 
 		_, _, watch, found := computer.GetIdentityPolicyByIdentity(id)
 		require.False(t, found)

--- a/pkg/policy/compute/compute_test.go
+++ b/pkg/policy/compute/compute_test.go
@@ -137,6 +137,51 @@ func TestRecomputeIdentityPolicy(t *testing.T) {
 	})
 }
 
+func TestIdentityManagerObserver(t *testing.T) {
+	testutils.GoleakVerifyNone(t, testutils.GoleakIgnoreCurrent())
+
+	t.Run("add triggers implicit recompute", func(t *testing.T) {
+		_, _, computer, idmgr := fixture(t)
+
+		targetID := identity.NumericIdentity(7)
+		id := identity.NewIdentity(targetID, labels.Labels{})
+
+		idmgr.Add(id)
+
+		obj := waitForEntry(t, computer, id, true)
+		assert.Equal(t, targetID, obj.Identity)
+	})
+
+	t.Run("remove deletes entry", func(t *testing.T) {
+		_, _, computer, idmgr := fixture(t)
+
+		targetID := identity.NumericIdentity(8)
+		id := identity.NewIdentity(targetID, labels.Labels{})
+
+		idmgr.Add(id)
+		waitForEntry(t, computer, id, true)
+
+		idmgr.Remove(id)
+		waitForEntry(t, computer, id, false)
+	})
+}
+
+func waitForEntry(t *testing.T, computer PolicyRecomputer, id *identity.Identity, want bool) Result {
+	t.Helper()
+	timeout := time.After(time.Second)
+	for {
+		obj, _, watch, found := computer.GetIdentityPolicyByIdentity(id)
+		if found == want {
+			return obj
+		}
+		select {
+		case <-watch:
+		case <-timeout:
+			t.Fatalf("timed out waiting for entry found=%v for identity %d", want, id.ID)
+		}
+	}
+}
+
 // computeFor adds an identity and waits for its initial policy to be committed.
 func computeFor(t *testing.T, computer PolicyRecomputer, idmgr identitymanager.IDManager, nid identity.NumericIdentity) *identity.Identity {
 	t.Helper()

